js/search/search.js

```diff
@@ -112,24 +111,36 @@ class Search {
     indexBatch(messages) {
         return new Promise((resolve, reject) => {
             if (!messages) {
+                log.send(logLevels.ERROR, 'Messages was not provided for batch indexing');
                 reject(new Error('Messages is required'));
                 return;
             }
 
-            if (!(JSON.parse(messages) instanceof Array)) {
-                reject(new Error('Messages must be an array'));
+            try {
+                let msg = JSON.parse(messages);
+                if (!(msg instanceof Array)) {
+                    log.send(logLevels.ERROR, 'Message must be an array batch indexing');
```

Comment:
Typo here. Should be ```Messages``` instead of ```Message```. Also, you can probably change the message to ```Batch Indexing: Messages must be an array```.